### ISSUE TITLE
Dupe assets during bundle for client/server

### ISSUE THREAD

**Issue by rtritto** [vikejs/vike#2034]
### Description

Do a build and `.css` static assets are duped in `dist/client/assets/static` and `dist/server/assets/static`.

### Repro (repo: [template-vike-solid-daisyui-hono](https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono) ~ branch: [dupe-assets-statics](https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono/tree/dupe-assets-statics))

- `git clone https://github.com/templates-ecosystem/template-vike-solid-daisyui-hono.git`
- `git checkout dupe-assets-statics`
- `cd template-vike-solid-daisyui-hono`
- `yarn`
- `yarn build` or `yarn vite build`
- see `dist/client/assets/static` and `dist/server/assets/static`

### Current
```console
├── dist
│   └── client
│   │    └── assets
│   │        └── static
│   │            └── layouts_styles-b42a6b88.R-5hASUW.css
│   └── server
│        └── assets
│            └── static
│                └── layouts_styles-b42a6b88.R-5hASUW.css
```

### Expected (draft)
```console
├── dist
│   ├── assets
│   │   └── static
│   │       └── layouts_styles-b42a6b88.R-5hASUW.css
│   └── client
│   └── server
```

**Comment by brillout** [comment]
@rtritto Up for digging into this? Let me know if you're interested and I'll tell you where the logic related to this is defined.

**Comment by rtritto** [comment]
I might take a look and do some research.

It would be handy to have more details/informations and to know where actually the problem is (vike/vike-node/vite/rollup); so as to help those who want to help (comments or PR).

**Comment by brillout** [comment]
`vike/node/plugin/plugins/buildConfig/fixServerAssets.ts`

**Comment by rtritto** [comment]
For the client build, the context (path) is `/dist/client` and on `rollupOptions.output.assetFileNames` output, the `assets/static/layouts_styles-b42a6b89.[hash][extname]` is correct. Changing `assets/static/layouts_styles-b42a6b89.[hash][extname]` to `../assets/static/layouts_styles-b42a6b89.[hash][extname]` doesn't work.

@brillout Do you have any idea?

**Comment by brillout** [comment]
If it's easy to support it, then let's support it. Otherwise let's add a `assertUsage()`.

**Comment by rtritto** [comment]
I don't know how to continue, the `rollupOptions.output.assetFileNames` uses the base path `/dist/client`

**Comment by brillout** [comment]
Let's add an `assertUsage()` then.

**Comment by rtritto** [comment]
@brillout still not fixed with vike@0.4.211 (I updated the repo)

**Comment by brillout** [comment]
Indeed
